pkg/controllers/provisioning/scheduling/machinetemplate.go

```diff
@@ -75,7 +75,9 @@ func (i *MachineTemplate) ToNode() *v1.Node {
 }
 
 func (i *MachineTemplate) ToMachine(owner *v1alpha5.Provisioner) *v1alpha5.Machine {
-	i.Requirements.Add(scheduling.NewRequirement(v1.LabelInstanceTypeStable, v1.NodeSelectorOpIn, lo.Map(i.InstanceTypeOptions, func(i *cloudprovider.InstanceType, _ int) string {
+	// Order the instance types by price and only take the first 200 of them to decrease the instance type size in the requirements
+	instanceTypes := lo.Slice(i.InstanceTypeOptions.OrderByPrice(i.Requirements), 0, 200)
```

Comment:
How did we come to 200?

Reply:
Heuristic, just a guess. I don't want to create such a small list that we can only have say like one spot instance type and a bunch of on-demand instances where we constrain it by spot when we do our launch.

Reply:
I know we talked about 100 but I have some concern about hitting the issue above with a smaller list ^

Reply:
At one point this value was 20, IIRC

Reply:
20? That one is much too small for sure. Do you have any specific issues with 200?

Reply:
Why is 20 instance types too small?

Reply:
For the reason I described above. AWS cloudprovider restricts on capacity type after receiving instance types from the scheduler, which means that if 19 of the top 20 instance types only have on-demand capacity availability, we will only take a single instance type when we do filtering on the cloudprovider side. The second reason is I can imagine a cloudprovider that doesn't have server-side fleet and wants a fuller array of the instance type options that the machine could launch with and 20 feels like it could be too unnecessarily constrained.